test/cookie.test.js

```diff
@@ -273,6 +273,42 @@ test('cookies signature', (t) => {
   })
 })
 
+test('cookies local signature', (t) => {
+  t.plan(6)
+  const fastify = Fastify()
+  const secret = 'bar'
+  fastify.register(plugin)
+
+  fastify.get('/test1', (req, reply) => {
+    reply
+      .setCookie('foo', 'foo', { signed: true, secret })
```

Comment:
I would add a more complex test:

```suggestion
      .setCookie('foo', 'foo', { signed: true, secret })
      .setCookie('bar', 'bar', { signed: true, anotherSecret })
```